app/src/main/java/com/nio/tcp/initializer/ServerSocketPortInitializer.java

```diff
@@ -0,0 +1,40 @@
+package com.nio.tcp.initializer;
+
+import com.nio.tcp.exception.PortInitializationException;
+import com.nio.tcp.singleton.SelectorSingleton;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+
+public class ServerSocketPortInitializer implements PortInitializer {
+
+    private static ServerSocketPortInitializer serverSocketPortInitializerInstance;
+
+    private final Selector selector;
+
+    public ServerSocketPortInitializer() {
+        this.selector = SelectorSingleton.getInstance();
+    }
+
+    public static ServerSocketPortInitializer getInstance() {
+        if (serverSocketPortInitializerInstance == null) {
+            serverSocketPortInitializerInstance = new ServerSocketPortInitializer();
+        }
+        return serverSocketPortInitializerInstance;
+    }
+
+    @Override
+    public void initPort(Integer port, String hostName) {
```

Comment:
wrong method name. Definitely, the server socket configured and bid int the method.